File: crates/g8-extractor/tests/fixtures/go/has_annotations.go
// PASS-detect fixture: Go file with G8 magic-comment annotations.

package example

// @g8.capability(name = "streaming-export", status = "in_flight", substrate = "export")
func ExportStream(w interface{}) error {
	return nil
}

// @g8.capability(name = "batch-export", status = "proposed", substrate = "export")
func ExportBatch(jobs []string) ([]byte, error) {
	return nil, nil
}

// @g8.convergence_test(for_capability = "streaming-export", scenario = "large-payload")
func TestStreamingExportLargePayload(t interface{}) {}

// @g8.intent(description = "Export pipeline functions", substrate = "export")
var _ = nil

// @g8.plan(title = "async-export", status = "Idea", substrate = "export")
func planAsyncExport() {}

File: crates/g8-extractor/tests/fixtures/ts/has_annotations.ts
// PASS-detect fixture: TypeScript file with G8 magic-comment annotations.

// @g8.capability(name = "batch-processor", status = "in_flight", substrate = "jobs")
export function processBatch(items: string[]): void {}

// @g8.convergence_test(for_capability = "batch-processor", scenario = "empty-batch")
describe("batch processor", () => {
  it("handles empty batch", () => {});
});

// @g8.intent(description = "Job processing utilities", substrate = "jobs")
const _intentMarker = null;

// @g8.plan(title = "retry-logic", status = "Idea", substrate = "jobs")
function _planMarker() {}
